lib/charms/vault_k8s/v0/vault_kv.py

```diff
@@ -379,6 +379,17 @@ def set_unit_credentials(
                 nonce,
             )
             return
+        standard_secret_id = (
+            secret.id[len("secret:") :] if secret.id.startswith("secret:") else secret.id
+        )
+        if standard_secret_id in credentials.get(nonce, ""):
+            logger.debug(
+                "Secret id has not changed, not updating the relation '%s:%d' for nonce %r",
+                relation.name,
+                relation.id,
+                nonce,
+            )
+            return
         credentials[nonce] = secret.id
```

Comment:
Shouldn't it be `credentials[nonce] = standard_secret_id`?

Reply:
No, it is correct the way it is.
We don't want to store the standard format.
The following is happening:
- The secret is created with an ID of the format `secret://model_uuid/standard_secret_id`
- ID is stored using `credentials[nonce] = secret.id`
- When the secret is updated later juju might return and ID with the format `secret://standard_secret_id` or `standard_secret_id` which can't be used to access the secret from a different model.

So with this workaround we keep the `secret://model_uuid/standard_secret_id` format.